Version 1.7.0

fingolfin opened the pull request:
There have been a ton of fixes and improvements since the last release, so we should make a release now.

Any objections or concerns @timholy @oxinabox or anyone else?

There are a couple other open issues that I think are easy enough to fix, but I don't think that's any reason to delay a release?


Draft of release notes for JuliaRegistrator bot:

----

Release notes:

## New features

- Add `sort!` method for unfrozen `LittleDict` ([#136](https://github.com/JuliaCollections/OrderedCollections.jl/pull/136))
- Add `mergewith` method for `LittleDict` ([#119](https://github.com/JuliaCollections/OrderedCollections.jl/pull/119))

## Bug fixes
- Fix `rehash!` for `OrderedDict` (for example `delete!(d, nothing)` could produce nonsense or even crash) ([#120](https://github.com/JuliaCollections/OrderedCollections.jl/pull/120))
- Avoid iterating twice over generators when there is an exception, at least in Julia >= 1.11 ([#137](https://github.com/JuliaCollections/OrderedCollections.jl/pull/137))
 
## Other changes
- Remove all uses `ccall` ([#121](https://github.com/JuliaCollections/OrderedCollections.jl/pull/121))
- Remove `convert` method for `Type{OrderedDict}`, the generic Julia method does the same, except for a deprecation warning our method produced ([#139](https://github.com/JuliaCollections/OrderedCollections.jl/pull/139))
- Several other janitorial changes

timholy commented:
I like the idea of making a release.

Did GitHub eliminate that link to see all the changes since the last release? It was a little broken anyway if one made non-increasing releases (e.g., 1.2.0 followed by 1.1.6), so perhaps they just dropped it.

fingolfin commented:
Such links exist on e.g. releases. One can also make one manually. E.g. to [list all changes since the last release 1.6.3](https://github.com/JuliaCollections/OrderedCollections.jl/compare/v1.6.3...master).

fingolfin commented:
I will make the release later today; I first want to prepare some manual release notes that summarize in a useful way (there are 43 commits but many of them could be summarized under "maintenance changes" or so). This would also be practice for a potential 2.0.0 release (which still needs a bunch of preparation before it could happen)
